src/structure_dump.cpp

```diff
@@ -414,46 +414,49 @@ void structure::set_chiP_from_file(h5file *file, const char *dataset, field_type
   }
 }
 
-binary_partition::binary_partition(int _proc_id) {
-  proc_id = _proc_id;
-  split_dir = NO_DIRECTION;
-  split_pos = 0.0;
-  left = NULL;
-  right = NULL;
+binary_partition::binary_partition(int _proc_id)
+    : proc_id(_proc_id), plane{NO_DIRECTION, 0.0} {}
+
+binary_partition::binary_partition(const split_plane &_split_plane,
+                                   std::unique_ptr<binary_partition> &&left_tree,
+                                   std::unique_ptr<binary_partition> &&right_tree)
+    : proc_id(-1), plane(_split_plane), left(std::move(left_tree)), right(std::move(right_tree)) {
+  if (!left || !right) { meep::abort("Binary partition tree is required to be full"); }
 }
 
-binary_partition::binary_partition(direction _split_dir, double _split_pos) {
-  split_dir = _split_dir;
-  split_pos = _split_pos;
-  proc_id = -1;
-  left = NULL;
-  right = NULL;
-}
+bool binary_partition::is_leaf() const { return !left && !right; }
+
+int binary_partition::get_proc_id() const { return proc_id; }
+
+const split_plane &binary_partition::get_plane() const { return plane; }
```

Comment:
Should these have `is_leaf()` and `!is_leaf()` assertions, respectively?

Reply:
Done.